rfc/0009-policy-report.md

```diff
@@ -0,0 +1,323 @@
+
+The `PolicyReport` resource has the following fields:
+
+* `apiVersion` and `kind`: these are defined by the CRDs
+* `metadata`: this is the usual `meta/v1.ObjectMeta` resource. In the
+  beginning we will set only the `name` field. We will do that by using the
+  following pattern: `polr-ns-<namespace name>`
+* `scope`: this is a `core/v1.ObjectReference` resource. We will use that to
+  reference the Kubernetes Namespace resource that has been analyzed
+* `scopeSelector`: we will not use that, because we are using the `scope` field
+* `summary`: this is an object defined by a CRD `PolicyReportSummary`. It's made
+  of the following fields:
+  * `pass`: count of policies whose requirements were met
+  * `fail`: count of policies whose requirements were not met
+  * `warn:`: we will not set it, it doesn't relate to us
+  * `error`: count of policies that could not be evaluated.
+    The policies could not be evaluated because of errors
+    (for example: a runtime panic of a broken policy)
+  * `skip`: count of policies that were not selected for evaluation.
+    We will allow some policies to be excluded from the background checks, more
+    on that later.
+* `results` this probably the most important field. It's a list of `PolicyReportResult`
+  objects
+
+The structure of the `PolicyReportResult` is flexible. This is how we are going
+to use its fields:
+
+* `source`: identifier for the policy engine that manages this report. We will
+  set that to be `kubewarden`
+* `policy`: this is the name of the policy that evaluated the resource. This
+  could be both a `ClusterAdmissionPolicy` or a `AdmissionPolicy`.
+  Because of that we cannot use the name of the policy, there could be clashes
+  between cluster and namespaced policy names. We will instead use the following
+  convention to build the contents of this field: `<[cap|ap]>-<policy name>`.
+  For example, a `ClusterAdmissionPolicy` named `provileged-containers` will
+  be identified via `cap-privileged-containers`. On the other hand, a `AdmissionPolicy`
+  called `required-labels` will be identified via `ap-required-labels`.
+* `rule`, `category`, `severity`, `scored`: we are going to leave these fields empty
+  because they do not relate to concepts we have
+* `timestamp`: the time the result was found
+* `result`: this is a string enum. These are the values we will be using:
+  `pass`, `fail` and `error`
+* `resources`: this is a list of `core/v1.ObjectReference` objects. We will
+  put only one object reference inside of this list. This is going to be a
+  reference to the resource that has been analyzed by the policy
+* `resourceSelector`: we are not going to use it, because we are using the `resources` field
+* `message`: this is used when the policy result is `fail` or `error`. It will hold
+  the output message provided by the policy
+* `properties`: we are not going to use this free-form dictionary in the beginning
```

Comment:
I wonder if we should add versioning for our usage of PolicyReports. And if so, if it could go here with a `version: v1`.